src/role_nextroomer.js

```diff
@@ -260,9 +260,16 @@ roles.nextroomer.settle = function(creep) {
   }
 
   let methods = [Creep.getEnergy];
-  if (creep.room.controller.ticksToDowngrade < 1500) {
+  if (creep.room.controller.ticksToDowngrade < 1500 || creep.room.controller.progress > creep.room.controller.progressTotal) {
     methods.push(Creep.upgradeControllerTask);
   }
+
+  let spawnCSs = creep.room.findPropertyFilter(FIND_MY_CONSTRUCTION_SITES, 'structureType', [STRUCTURE_SPAWN]);
+  let spawns = creep.room.findPropertyFilter(FIND_MY_STRUCTURES, 'structureType', [STRUCTURE_SPAWN]);
+  if (spawns.length === 0 && spawnCSs.length > 0) {
+    methods.push(Creep.constructTask);
+  }
```

Comment:
K, let's discuss this.

The idea behind the current prioritization:
 - RCL < 3 the room is useless, no tower, not enough energy from proper creeps (tiny creep handling is annoying, happened a lot of time that a big nextroomer was waiting for a tiny harvester queued at the source)
 - RCL 3 the tower should be built and filled to have some kind of defense
 - RCL 4 is where we want to get quickly so that we have a storage and the proper base process starts

I'm fine to have the spawn build early.

On my (a higher) GCL you have a couple of rooms sending nextroomer, which are 50 parts creeps which suck up the sources pretty fast and fill up the controller. Also in case other players interrupt you build up process we should take care of that.